usl_models/tests/flood_ml/mock_dataset.py

```diff
@@ -77,3 +85,47 @@ def generator():
     if batch_size:
         dataset = dataset.batch(batch_size)
     return dataset
+
+
+def mock_prediction_dataset(
+    params: model_params.FloodModelParams,
+    height: int = 100,
+    width: int = 100,
+    batch_count: int = 1,
+    batch_size: int = 1,
+    n: int = 0,
+) -> tf.data.Dataset:
+    """Constructs a dataset of random mock prediction data.
+
+    Args:
+        params: Flood model params
+        height: spatial map height
+        width: spatial map width
+        batch_count: number of batches to produce
+        batch_size: size of each batch
+        n: optional number of timesteps.
+            Required for producing data for use with `model.call_n`.
+
+    Returns:
+        The TF dataset.
+    """
+    input_sig = input_signature(params, height, width, n)
+    metadata_sig = metadata_signature()
+
+    def generator():
+        """Generate random tensors."""
+        for i in range(batch_size * batch_count):
+            yield flood_model.FloodModel.Input(
+                geospatial=tf.random.normal(shape=input_sig["geospatial"].shape),
+                temporal=tf.random.normal(shape=input_sig["temporal"].shape),
+                spatiotemporal=tf.random.normal(
+                    shape=input_sig["spatiotemporal"].shape
+                ),
+            ), dict(feature_chunk=str(i), rainfall=i)
+
+    dataset = tf.data.Dataset.from_generator(
+        generator=generator, output_signature=(input_sig, metadata_sig)
+    )
+    if batch_size:
```

Comment:
Won't this always be true, unless we set batch size 0? It seems strange to take 0 if we don't want to batch -- should this be optional instead?

Reply:
Sure, we can use None instead.